### PR TITLE
refactor GitHub output command to current version

### LINKED ISSUE #220: GH Actions: replace all `set-output` with proper step outputs
## WHAT

`set-output` is deprecated and using it will log a warning in GH Actions:
```yaml
echo "::set-output name=commit::$(SOME_COMMAND)"
```
will result in a warning: 
> The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

this needs to be done for all our workflows.

### PR DESCRIPTION
**Pull request by gcs14**
## WHAT

Refactor GitHub output command to current version

## WHY

The current command has been deprecated by GitHub Actions and will be disabled soon.

Closes #220

### CODE REVIEW

**Review by paullatzelsperger — CHANGES_REQUESTED** [review]
If I understand the documentation (especially the [examples](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#environment-files)) correctly, the curly braces aren't needed. `echo` will print literally whatever's written between the quotes (expanding variables and commands). So instead of 
```
echo "{commit}={$(git rev-parse HEAD)} >> $GITHUB_OUTPUT
```
you would need to write
```
echo "commit=$(git rev-parse HEAD) >> $GITHUB_OUTPUT
```

File: .github/workflows/draft-new-release.yaml [hunk 1]
```diff
@@ -68,7 +68,7 @@ jobs:
           git add CHANGELOG.md gradle.properties $(find charts -name Chart.yaml) $(find charts -name README.md)
           git commit --message "Prepare release ${{ github.event.inputs.version }}"
 
-          echo "::set-output name=commit::$(git rev-parse HEAD)"
+          echo "{commit}={$(git rev-parse HEAD}" >> $GITHUB_OUTPUT
```
Comment (paullatzelsperger) [inline .github/workflows/draft-new-release.yaml:71]: forgot the closing brace

Comment (gcs14) [reply .github/workflows/draft-new-release.yaml:71]: Ah I see, I'll update those changes. Thank you.

**Comment by paullatzelsperger** [comment]
@gcs14 Looks good now, thanks. For future contributions, please re-request a review after you've incorporated all change-requests, as per our contribution guidelines. That'll help reviewers keep an eye on their work.

**Review by paullatzelsperger — APPROVED** [review]

**Pull request merged**